api-extractor/src/definitions/ApiDocumentation.ts

```diff
@@ -59,6 +58,14 @@ export interface IScopePackageName {
   package: string;
 }
 
+/**
+ * A dependency for ApiDocumentation constructor that abstracts away the function 
+ * of resolving an API definition reference.
+ */
+export interface IReferenceResolver {
+  resolve(apiDefinitionRef: IApiDefinitionReference, reportError: (message: string) => void): IDocItem;
```

Comment:
We need some documentation about the behavior of reportError:

- If there is an error, what does the function return?  undefined?
- If the function returns a result, can reportError() ever be called also?
- Can reportError() be called twice?
- If the function returns undefined, is it guaranteed that reportError() is called at least once?

These details would be important to someone implementing or consuming the interface.